graph/graph.go

```diff
@@ -231,11 +233,46 @@ func checkVisibility(target labels.Label, visibilities []string) bool {
 	return false
 }
 
-func writeFormattedBuildFile(buildFile *build.File, out io.Writer, format string) error {
+type nopCloser struct {
+	io.Writer
+}
+
+func (nopCloser) Close() error { return nil }
+
+// writeFormattedBuildFile writes a build file to the given writer if puku has made meaningful changes.
+//
+// See the comment on outputFormattedBuildFile for more details.
+func writeFormattedBuildFile(buildFile *build.File, out io.Writer, format string, skipRewriting bool) error {
+	outFn := func() (io.WriteCloser, error) {
+		return nopCloser{out}, nil
+	}
+	return outputFormattedBuildFile(buildFile, outFn, format, skipRewriting)
+}
+
+// saveFormattedBuildFile writes a build file to disk if puku has made meaningful changes.
+//
+// See the comment on outputFormattedBuildFile for more details.
+func saveFormattedBuildFile(buildFile *build.File, skipRewriting bool) error {
+	outFn := func() (io.WriteCloser, error) {
+		return os.Create(buildFile.Path)
+	}
```

Comment:
Found this a bit hard to follow. Can't you just create the file here and pass in the writer to `outputFormattedBuildFile()` rather than using a closure? This function can defer the close.

Reply:
I've added a comment to explain why it's like this.